Feat/updt typography

bholaday opened the pull request:
Code review for the new typography items for the new pattern library:

- modify heading sizes, fonts, and styles
- new text highlight (BEM)
- new pull quote styles (BEM)
- modify ordered and unordered lists
  - bullets and numbers are now campusGold
  - bullets are larger
- add `list` block that applies the default list styles
  - add `roman-upper` modifier that changes the list-style of ordered lists to use the following hierarchy:
    1. upper roman numeral
    2. upper alpha
    3. decimal
    4. lower alpha
    5. lower roman numeral

    _Note: This structure was created based on lists used on the Purdue policies site._

An example of the changes can be found on our [test page](https://dev.www.purdue.edu/pmmtest/templates/testing/typography/templates/patterns/typography.html).

bholaday commented on components/html/modules/content-patterns/_typography.html:
```diff
@@ -0,0 +1,50 @@
+				<p>Obcaecati aliquid, aspernatur exercitationem. Explicabo, impedit repudiandae maxime. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Debitis impedit suscipit cum ea cumque reiciendis laudantium sed, aut unde optio molestiae non aliquam officia aliquid est earum minus qui nulla. Lorem ipsum dolor sit amet, consectetur adipisicing elit.  Commodi, quod earum voluptatem dolorem aliqu ...
+				<h4>H4 Archivo Bold 24px Caps #98700d</h4>
+				<p>Dolor sit amet, consectetur adipisicing elit. Obcaecati aliquid, aspernatur exercitationem. Explicabo, impedit repudiandae maxime. Commodi, quod earum voluptatem dolorem aliquam omnis repellendus vel quisquam similique dolores, esse maxime. <a href="#">Lorem ipsum dolor sit amet, consectetur adipisicing elit.</a> Debitis impedit suscipit cum ea cumque reiciendis laudantium sed, aut unde op ...
+				<aside class="highlight">
```
I wasn't sure what class name to give this style - the gray text highlight

jacobvenable replied:
I think that name works just fine.

bholaday commented on components/html/modules/content-patterns/_typography.html:
```diff
@@ -0,0 +1,50 @@
+				<p>Explicabo, impedit repudiandae maxime. Commodi, quod earum voluptatem dolorem aliquam omnis repellendus vel quisquam similique dolores, esse maxime.</p>
+				<h6>H6 Archivo Bold 18px Caps #000000</h6>
+				<p>Aliquam officia aliquid est earum minus qui nulla. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Obcaecati aliquid, aspernatur exercitationem. Commodi, quod earum voluptatem dolorem aliquam omnis repellendus vel quisquam similique dolores, esse maxime.</p>
+				<aside class="pullQuote">
```
This would represent the default style for a pull quote - a gold bar on the left.

bholaday commented on components/html/modules/content-patterns/_typography.html:
```diff
@@ -0,0 +1,50 @@
+					<p class="pullQuote__source">- Archivo Narrow Bold 18px #98700d</p>
+				</aside>
+				<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Sed quidem similique, aspernatur illum tempora, nostrum pariatur eos consequuntur atque minus fugit deleniti illo reprehenderit architecto quis ipsam debitis asperiores temporibus!</p>
+				<aside class="pullQuote pullQuote-quote">
```
This modifer replaces the gold bar with a big smart quote.

bholaday commented on components/css/mixins/_colors.scss:
```diff
@@ -15,6 +15,10 @@ $color-mackeyOrange:#ff9b1a;
 $color-yellowWalk:#ffd100;
 $color-fountainRunTeal:#29a592;
 $color-pappysPurple:#ad1f65;
```
Garrett defined these grays a while ago. I think they should make their way into production.

bholaday commented on components/css/global/_base.scss:
```diff
@@ -10,77 +10,66 @@ body{
 	body{
```
Throughout this file, I modified the font sizes to use our function and consolidated redundancies.

bholaday commented on components/css/content/_contentPage.scss:
```diff
@@ -52,11 +52,41 @@
 					width:100%;
 					margin-bottom:20px;
 				}
```
If we're going to use colored bullets and numbers we have to do some trickery. I havent tested fallbacks for this yet.

bholaday commented on components/css/content/_contentPage.scss:
```diff
@@ -52,11 +52,41 @@
+						display: inline-block;
+						font-size: calc-toRem(33);
+						line-height: 0;
+						top: 5px;
```
I may need to make these (r)em so they'll scale nicely.

jacobvenable replied:
Yep! I think using calc-toRem(5) in its place should work.

jobokai commented:
Maybe it's just me, but it feels awkward going from the super heavy H1 to the regular H2 to a bold H3. Just my two cents.

bholaday commented:
@jobokai Thanks for your comment. That's why we put it up. I'll pass it along to our designer.

jobokai commented:
@bholaday I think everything else looks really good and clean. Do the ordered and un-ordered lists handle multi-line cleanly? IE they don't wrap down to full width on a new line but rather keep the indentation and stay aligned cleanly.

bholaday commented:
@jobokai Thanks. I havent done a lot of testing yet. (the lists will need a good amount of that since its a big change) This pull request was mainly to get input on the direction of things and course correct before I get too far into it. ;-)

bholaday commented:
Wrapping is decent. There's a bit of an indent on the numbered lists. Inspecting, it actually looks like the first line is actually the oddity.

![screen shot 2018-02-13 at 12 51 38 pm](https://user-images.githubusercontent.com/5982996/36165179-a81bebd0-10bc-11e8-9cb1-d0825fdb93d8.png)

jobokai commented:
That's really odd behavior for the ordered list.... Yea the direction and updates look really great and I think it's in the right direction :)

bholaday commented on components/css/global/_base.scss:
```diff
@@ -10,77 +10,66 @@ body{
 	color:$color-headlineGold;
-    font-size:36/16 * 1em;
+    font-size: calc-toEm(38);
     line-height:1.1em;
```
I forgot to make this uppercase. This will be changed in a future update.

jobokai commented:
One thing to maybe take a look at are horizontal rules? I'm not sure if there's existing styles for them

bholaday commented:
Good idea. We aren't using them at all (that I know of), but we should give them a style that fits with the brand.

jacobvenable commented on components/css/content/_contentPage.scss:
```diff
@@ -52,11 +52,41 @@
+					}
+				}
+				ol {
+					counter-reset: li;
```
Using `counter-reset` and `conter-increment` is pretty slick. This could break how ordered lists display in older browsers though. Perhaps just wrap these properties in feature queries, `@supports(counter-reset:li)`, so that numbered lists can be displayed properly in older browsers. This would mean that numbers would not be colored in IE, https://caniuse.com/#search=%40support , but that is ok with me.

bholaday replied:
Done. I lumped all the gold styles for both lists in the feature query. Technically I could make the bullets gold regardless of counter support, but for consistency, I opted to have all or nothing. I'll confirm with Garrett.

jacobvenable replied:
Do you mean the bullets or the numbers? If the bullets are ok regardless of counter support, those don't have to be placed into the feature query.

jacobvenable commented on components/css/global/_base.scss:
```diff
@@ -10,77 +10,66 @@ body{
 	padding-bottom:0.25em;
-	font-family:$font-archivo-narrow;
-	font-weight:bold;
+	font-family:$font-champion-welter;
```
I don't think our template is currently loading in champion. We should probably add this to the _head HTML module before merging.

bholaday replied:
This is kind of up in the air anyway until we get the font situation figured out anyway.

jacobvenable commented on components/css/partials/_highlight.scss:
```diff
@@ -0,0 +1,31 @@
+        color: inherit;
+        text-transform: uppercase;
+        font-size: calc-toRem(22);
+        font-weight: bold;
```
I would go ahead and declare the font-family for this selector as well just to ensure that it appears as intended.

jacobvenable commented on components/css/partials/_highlight.scss:
```diff
@@ -0,0 +1,31 @@
+.highlight {
+    margin: 1em 0;
+    padding-left: 2em;
```
Perhaps we should consider reducing this padding on smaller devices to save a little bit of horizontal space.

jacobvenable commented on components/css/global/_base.scss:
```diff
@@ -10,77 +10,66 @@ body{
+	font-family:$font-champion-welter;
 	color:$color-headlineGold;
-    font-size:36/16 * 1em;
+    font-size: calc-toEm(38);
```
Although they don't currently, perhaps the heading font-sizes should change based on the window width. The main heading could there possibly being a problem with larger words in a `h1` bleeding off mobile devices.

bholaday replied:
I'll loop Garrett in and have him mock something up for mobile (and maybe tablet)

jacobvenable commented on components/css/global/_base.scss:
```diff
@@ -10,77 +10,66 @@ body{
 }
 h1{
 	padding-top:0;
 	padding-bottom:0.25em;
```
I made this mistake when I originally setup this file, but I think these should be margins instead of padding. Since margins collapse on each other, it would be easier to manage vertically spacing elements instead of handling it via more complex selectors, such as `h1+h2`. We'll have to be cautious that implementing margins instead of padding doesn't affect templates.

bholaday replied:
Yes, I wondered about that and I agree. I'll comment these sections out and switch to margins and test.

jacobvenable commented on components/css/partials/_pullQuote.scss:
```diff
@@ -0,0 +1,50 @@
+.pullQuote {
```
Possibly rename block to `quote`?

jacobvenable replied:
Maybe since both of these styles are reserved for quotes, we should just keep in a separate block.

bholaday commented:
I pushed up some changes last night that include:

- added Champion (temporary for now) to style H1 correctly
- changed class `pullQuote` to `quote`
- minor ruleset changes for spacing, font, etc.
- added mobile styles to the highlight and quote
  - padding is reduced to give more room at 600px and below
- changed heading padding to margins so they can collapse on other elements and reduced the top margin for consistency between all heading levels (per Garrett's desire)
- major overhaul of the way lists were being built
  - a feature query only displays gold ordered lists if the browser supports it, gold bullets are displayed regardless
  - unordered lists have three levels of bullets - solid circle, hollow circle, solid square
  - ordered lists, if supported, have three levels of "numbers" - digits, lower-alpha chars, lower roman numerals

bholaday commented on components/css/global/_base.scss:
```diff
@@ -10,103 +10,101 @@ body{
+// }
 p{
 	margin:0 !important;
 	padding:7.5px 0;
```
Should we also change this padding to margin?

jacobvenable replied:
Yep, I would say so.

jacobvenable commented on components/css/content/_contentPage.scss:
```diff
@@ -53,40 +53,119 @@
+							&::before {
+								background-color: #fff;
+								border-radius: 50%;
+								border: calc-toRem(2) solid $color-campusGold;
```
Noice, I like this.

jacobvenable commented on components/css/content/_contentPage.scss:
```diff
@@ -53,40 +53,119 @@
+					ol > li {
+						&::before {
+							position: relative;
+							color: $color-campusGold;
```
Just realized that since this needs to be readable, this should be `$color-headerGold` since the campus gold doesn't provide enough contrast on white.

jacobvenable commented on components/css/content/_contentPage.scss:
```diff
@@ -53,40 +53,119 @@
+						> li {
+							&::before {
+								counter-increment: l1;
+								content: counter(l1)".";
```
Thoughts on also applying `font-weight:700;`?

bholaday replied:
Garrett agrees, done.

jacobvenable commented on components/css/global/_base.scss:
```diff
@@ -74,28 +83,28 @@ h1,h2,h3,h4,h5,h6{
+// }
+// h1+h2{
+// 	padding-top:0.15em;
+// }
```
Are these necessary anymore since the headers now default to the largest margin to determine the distance between them?

bholaday replied:
Gone.

jacobvenable commented on components/css/partials/_quote.scss:
```diff
@@ -1,28 +1,30 @@
     }
 
-    &-quote {
+    &-quotemark {
```
Maybe rename modifier to `mark`?

jacobvenable commented on components/css/global/_base.scss:
```diff
@@ -1,163 +1,124 @@
+
+
+h1, h2, h3, h4, h5, h6 {
+	a:active, a:hover, a:link, a:visited {
```
Although it doesn't matter in this case, lets make sure the order is always:
`a:link, a:visited, a:hover, a:active`

bholaday replied:
Done

jacobvenable commented on components/css/mixins/_all.scss:
```diff
@@ -1,4 +1,5 @@
-@import 'fonts';
\ No newline at end of file
+@import 'fonts';
+@import 'h';
```
Can we rename this `heading` or `headings`? I see you already have a partial named that, but I think this works since it's in a separate folder.

bholaday replied:
Yes

jacobvenable commented on components/html/modules/content-patterns/_typography.html:
```diff
@@ -3,79 +3,79 @@
-				<h6>H6 Archivo Bold 18px Caps #000000</h6>
+				<h6>H6 Archivo Narrow Bold 1.125em Caps #000000</h6>
 				<p>Aliquam officia aliquid est earum minus qui nulla. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Obcaecati aliquid, aspernatur exercitationem. Commodi, quod earum voluptatem dolorem aliquam omnis repellendus vel quisquam similique dolores, esse maxime.</p>
 				<aside class="quote">
```
Let's go ahead and apply the `.quote` style to the `blockquote` element as its base. We may need override some default bootstrap styles.

jacobvenable commented on components/css/partials/_quote.scss:
```diff
@@ -14,7 +14,7 @@
+    &-mark {
         padding-left: 4rem;
 
         &::before {
```
So I just realized that screen readers will read "left double quotation mark" separately in front of the quote title/content for this modifier. I'm not quite sure how to handle this. We could replace the content with the FontAwesome icon, https://fontawesome.com/icons/quote-left?style=solid , but that still has the issue of the screen reader coming upon a separate element that it can't read.

We could use an SVG background image and make the the content property and empty string http://jsfiddle.net/6WAtQ/ . We'd have to set the height and width equal to rems instead of the font-size as well. That way, the screen reader would have nothing to read in the first place.

Thoughts?

jacobvenable replied:
I decided to cook up a couple previews of what using SVG inline would look like:

## Outline of the Archivo Narrow Curly Quote:
https://jsfiddle.net/jvenabl/hj8p26zv/23/
**Pros:**
- Easier to position

**Cons:**
- Can't change quote font as easily since it's an outline

## Actual text in the SVG:
https://jsfiddle.net/jvenabl/zrf1jffj/39/
**Pros:**
- Can easily change the quote font by just changing the font-family.

**Cons:**
- If you update the font, you need to mess with text element's position and possibly font size (it's kind of a pain).

bholaday replied:
What about a "speech" media query?

```
@media speech {
    .quote-mark::before { {
        display: none;
        visibility: hidden;
    }
}
```

bholaday replied:
No go. Doesnt work.

bholaday replied:
Just for reference I played around with a few other ideas with no luck: `speak: none;` and `speak-as: no-punctuation;`

I'd opt for option one since I don't think we're going to be changing it often, and even if we did, I don't think the process would be too hard.

bholaday replied:
I've implemented option 1. It will be reflected in the next push.

jacobvenable commented on components/html/modules/content-patterns/_typography.html:
```diff
@@ -3,79 +3,79 @@
-					<h3 class="quote__content">"Archivo Narrow Bold 24px #2e2e2e Consectetur adipisicing elit. At illum esse quas sunt, reiciendis magnam itaque aut exercitationem et, voluptatum laboriosam expedita, quia officia ipsum nemo molestias alias sequi vero."</h3>
-					<span class="quote__source">- Archivo Narrow Bold 18px #98700d</span>
+				<aside class="quote quote-mark">
+					<h3 class="quote__content">"Archivo Narrow Bold 1.5em #2e2e2e Consectetur adipisicing elit. At illum esse quas sunt, reiciendis magnam itaque aut exercitationem et, voluptatum laboriosam expedita, quia officia ipsum nemo molestias alias sequi vero."</h3>
```
Although this is just an example, we probably should use different element for the quote content, such as a `<p>`.

bholaday replied:
Good catch. I think that just got copied and never changed.

jacobvenable commented on components/html/modules/content-patterns/_typography.html:
```diff
@@ -3,79 +3,79 @@
-				<aside class="quote quote-quotemark">
-					<h3 class="quote__content">"Archivo Narrow Bold 24px #2e2e2e Consectetur adipisicing elit. At illum esse quas sunt, reiciendis magnam itaque aut exercitationem et, voluptatum laboriosam expedita, quia officia ipsum nemo molestias alias sequi vero."</h3>
-					<span class="quote__source">- Archivo Narrow Bold 18px #98700d</span>
+				<aside class="quote quote-mark">
```
Let's go ahead and change this to a `<blockquote>` element.

bholaday replied:
Done.

jacobvenable commented on components/html/modules/content-patterns/_typography.html:
```diff
@@ -3,79 +3,79 @@
+					<li>Arial Regular</li>
+					<li>Arial Regular</li>
+					<li>Arial Regular</li>
 					<li>Nested<ol>
```
Voiceover reads the nested list-item elements differently than those that do not contain another ordered list. For example, it reads this element element "4" "period" "Nested", each being read separately. I'm not really sure how to handle this yet.

bholaday commented:
A few recent updates:

- updated the quote mark to use svg instead of content so screen readers wont read aloud "double quotes"
- added the default quote style to the blockquote element

Issues:
- still not sure how to handle the screen reading with nested lists

bholaday commented:
@jengarza @jacobvenable Is this ready to merge? 

Currently, I have a partial containing the classes that use the heading styles, but they aren't being referenced. Do we want to include those in the build?

Finally, when this is ready to merge, I'll first need to link up the new files to each template page type. (Currently, I am only loading them into the content page. (since the pattern page itself is a content page type)

jengarza commented:
@bholaday I'm good with the changes.

jacobvenable commented:
@bholaday I would say that we should include that heading classes partial as well.

I'm not sure how to handle the remaining issue of how screen readers handle the nested lists. Its not a huge deal but certainly not optimal.

bholaday commented:
Yes, let's plan to merge this soon and continue to look for solutions to the odd reading of nested lists.

jacobvenable commented:
Just pushed a small update containing the falling changes:

- fix the position of the numbers used on ordered lists
- create list mixins that contain styles for various list type so they can be reused on default `list/list-item` elements and also the `list` block.
- create new `list` block that applies the same default styles via a class. This also includes a `roman-upper` modifier that changes the hierarchy of nested, ordered lists. See updated pull-request description for the new hierarchy.
